In [1]:
import pandas as pd

df=pd.read_csv("../data/raw/IPL_Ball_by_Ball_2008_2022.csv")

##### We will look if our batter and bowler names exist as per our assumptions or not

In [2]:
df['batter'].unique()

array(['YBK Jaiswal', 'JC Buttler', 'SV Samson', 'D Padikkal',
       'SO Hetmyer', 'R Ashwin', 'R Parag', 'TA Boult', 'OC McCoy',
       'WP Saha', 'Shubman Gill', 'MS Wade', 'HH Pandya', 'DA Miller',
       'V Kohli', 'F du Plessis', 'RM Patidar', 'GJ Maxwell', 'MK Lomror',
       'KD Karthik', 'Shahbaz Ahmed', 'PWH de Silva', 'HV Patel',
       'JR Hazlewood', 'Q de Kock', 'KL Rahul', 'M Vohra', 'DJ Hooda',
       'MP Stoinis', 'E Lewis', 'KH Pandya', 'PVD Chameera', 'PK Garg',
       'Abhishek Sharma', 'RA Tripathi', 'AK Markram', 'N Pooran',
       'Washington Sundar', 'R Shepherd', 'J Suchith', 'B Kumar',
       'Umran Malik', 'JM Bairstow', 'S Dhawan', 'M Shahrukh Khan',
       'MA Agarwal', 'LS Livingstone', 'JM Sharma', 'PN Mankad',
       'PP Shaw', 'DA Warner', 'MR Marsh', 'RR Pant', 'SN Khan',
       'R Powell', 'AR Patel', 'SN Thakur', 'Kuldeep Yadav',
       'Ishan Kishan', 'RG Sharma', 'D Brevis', 'Tilak Varma', 'TH David',
       'Ramandeep Singh', 'DR Sams', 'RD Gaikwa

In [3]:
df['bowler'].unique()

array(['Mohammed Shami', 'Yash Dayal', 'LH Ferguson', 'Rashid Khan',
       'HH Pandya', 'R Sai Kishore', 'TA Boult', 'M Prasidh Krishna',
       'YS Chahal', 'OC McCoy', 'R Ashwin', 'Mohammed Siraj',
       'JR Hazlewood', 'GJ Maxwell', 'Shahbaz Ahmed', 'HV Patel',
       'PWH de Silva', 'Mohsin Khan', 'PVD Chameera', 'KH Pandya',
       'Avesh Khan', 'Ravi Bishnoi', 'AS Joseph', 'LS Livingstone',
       'Arshdeep Singh', 'K Rabada', 'NT Ellis', 'Harpreet Brar',
       'B Kumar', 'Washington Sundar', 'Fazalhaq Farooqi', 'J Suchith',
       'Umran Malik', 'R Shepherd', 'DR Sams', 'HR Shokeen', 'JJ Bumrah',
       'M Markande', 'RP Meredith', 'Ramandeep Singh', 'KK Ahmed',
       'A Nortje', 'SN Thakur', 'MR Marsh', 'Kuldeep Yadav', 'AR Patel',
       'Mukesh Choudhary', 'Simarjeet Singh', 'MJ Santner', 'M Pathirana',
       'MM Ali', 'PH Solanki', 'S Kaul', 'MK Lomror', 'UT Yadav',
       'TG Southee', 'SP Narine', 'CV Varun', 'AD Russell', 'N Rana',
       'JO Holder', 'K Gowtham', 'M

In [4]:
[name for name in df['batter'].unique() if 'Villiers' in str(name)]

['AB de Villiers']

In [5]:
[name for name in df['bowler'].unique() if "Bravo" in str(name)]

['DJ Bravo']

In [6]:
# now we can define our targets 

batter="AB de Villiers"

bowlers=[
    "JJ Bumrah",
    "DJ Bravo",
    "DW Steyn",
    "SL Malinga",
    "R Ashwin",
    "RA Jadeja",
    "SP Narine"

]

In [7]:
filtered_df=df[
    (df['batter']==batter) &
    (df['bowler'].isin(bowlers))
]

filtered_df.head()

,ID,innings,overs,ballnumber,batter,bowler,non-striker,extra_type,batsman_run,extras_run,total_run,non_boundary,isWicketDelivery,player_out,kind,fielders_involved,BattingTeam
18490,1254115,1,12,3,AB de Villiers,SP Narine,GJ Maxwell,NaN,1,0,1,0,0,NaN,NaN,NaN,Royal Challengers Bangalore
18492,1254115,1,12,5,AB de Villiers,SP Narine,GJ Maxwell,NaN,0,0,0,0,0,NaN,NaN,NaN,Royal Challengers Bangalore
18493,1254115,1,12,6,AB de Villiers,SP Narine,GJ Maxwell,NaN,4,0,4,0,0,NaN,NaN,NaN,Royal Challengers Bangalore
18500,1254115,1,14,1,AB de Villiers,SP Narine,GJ Maxwell,NaN,2,0,2,0,0,NaN,NaN,NaN,Royal Challengers Bangalore
18501,1254115,1,14,2,AB de Villiers,SP Narine,GJ Maxwell,NaN,0,0,0,0,1,AB de Villiers,bowled,NaN,Royal Challengers Bangalore


In [8]:
filtered_df.shape

(418, 17)

##### from the shape we can interpret abd faced thsese bowlers for 418 deliveries

In [9]:
filtered_df['batter'].unique()

array(['AB de Villiers'], dtype=object)

In [10]:
filtered_df['bowler'].unique()

array(['SP Narine', 'DJ Bravo', 'JJ Bumrah', 'R Ashwin', 'RA Jadeja',
       'SL Malinga', 'DW Steyn'], dtype=object)

#### Just double checking if our filtered df shows the correct names

In [13]:
filtered_df.to_csv("../data/processed/add_vs_bowlers.csv",index=False)